homeassistant/components/camera/__init__.py

```diff
@@ -0,0 +1,1133 @@
+
+            all_subfiles = sorted(
+                all_subfiles,
+                key=lambda x: os.path.getctime(
+                    os.path.join(event_data['fullPath'], x)),
+                reverse=False)
+
+            for image_file in all_subfiles:
+                full_image_path = os.path.join(
+                    event_data['fullPath'],
+                    image_file)
+
+                image_data = {}
+                image_data['fileName'] = image_file
+                image_data['path'] = (
+                    base_path +
+                    os.path.sep +
+                    event_dir +
+                    os.path.sep +
+                    image_file)
+
+                image_data['url'] = (
+                    'api/saved_camera_image/' +
+                    self.entity_id +
+                    '?image_path=' +
+                    image_data['path'])
+
+                image_data['time'] = datetime.datetime.fromtimestamp(
+                    os.path.getctime(
+                        full_image_path)).strftime('%Y-%m-%d %H:%M:%S')
+                event_data['images'].append(image_data)
+
+            if len(event_data['images']) > 0:
+                thumb_index = math.floor(len(event_data['images'])/2)
+                event_data['thumbUrl'] = (
+                    event_data['images'][thumb_index]['url'])
+
+            events_data.append(event_data)
+            count += 1
+
+        return events_data
+
+    def record_stream(self):
+        """ Records individual frames to disk for a period of time """
+        if self.is_recording:
+            return
+        else:
+            self._is_recording = True
+
+        self.update_ha_state(True)
+        self.send_recording_state()
+
+        rec_dir = (
+            'recording-' +
+            datetime.datetime.now().strftime('%Y-%m-%d_%H-%M-%S'))
+
+        rec_dir_full = os.path.join(self.recording_images_path, rec_dir)
+
+        if not os.path.isdir(rec_dir_full):
+            os.makedirs(rec_dir_full)
+
+        recording_length = DEFAULT_RECORDING_SECONDS
+        recording_started = datetime.datetime.now()
+        recording_time = 0
+        while (
+                recording_time < recording_length
+                and self._is_recording):
+
+            new_file_name = (
+                'recording_image-' +
+                datetime.datetime.now().strftime(
+                    '%Y-%m-%d_%H-%M-%S-%f') + '.jpg')
+
+            new_file_path = os.path.join(rec_dir_full, new_file_name)
+
+            response = self.get_camera_image()
+            open(new_file_path, 'wb').write(response.content)
+
+            recording_time = (
+                datetime.datetime.now() - recording_started).total_seconds()
+
+        self._is_recording = False
+        self.update_ha_state(True)
+        self.send_recording_state()
+
+    def stop_recording(self):
+        """ Stop recording camera stream """
+        self._is_recording = False
```

Comment:
Is this enough to stop recording or is this method supposed to be overwritten by platforms?